Replace rappdirs by tools::R_user_dir()

Bisaloo opened the pull request:
I see that this change was made in other r-lib repos so I thought I'd suggest it here as well.

`tools::R_user_dir()` was introduced in R 4.0 so safe to use with the tidyverse R version support policy/

Bisaloo committed:
Replace rappdirs by tools::R_user_dir()

Bisaloo committed:
Use nzchar() to test for path existence

hadley commented:
I think this probably needs a bit more thought than in other packages, because we'd need to make sure that this directory eventually gets cleaned up.

hadley commented:
httr2 has `cache_disk_prune()` which is called automatically on load, and ensures that all tokens > 30 days old are deleted.

hadley committed:
Merged origin/main into Bisaloo-rm-rappdirs

hadley committed:
Handle both old and new

hadley committed:
Inline function so we can move rappdirs to suggests

hadley committed:
Respect R_USER_CACHE_DIR

hadley committed:
Work on windows

hadley committed:
Always clear both

hadley committed:
Merge branch 'main' into Bisaloo-rm-rappdirs

hadley committed:
Move news bullet to correct place

hadley committed:
Fix flaky legacy path test on Windows by comparing filesystem equivalence instead of path strings

hadley committed:
Merge remote-tracking branch 'origin/main' into Bisaloo-rm-rappdirs

# Conflicts:
#	DESCRIPTION
#	NEWS.md

hadley committed:
Normalize legacy OAuth cache path with shortPathName() on Windows

Compares the inlined legacy cache path against `rappdirs::user_cache_dir()`
by converting both to their canonical short (8.3) form on Windows, since
rappdirs uses the CSIDL API which can return a short-form home directory
while our env-var based path uses the long form.

hadley committed:
Simplify given rlang re-hashing

hadley committed:
Fix test

hadley committed:
Merge commit '981db5f36334865a069c764e0f8799948ab49976'

hadley committed:
Revert spurious changes
